src/_P159_LD2410.ino

```diff
@@ -0,0 +1,245 @@
+          options[option] = concat(Plugin_159_valuename(P159_OUTPUT_STATIC_DISTANCE_GATE_index, true),
+                                   (option - P159_OUTPUT_STATIC_DISTANCE_ENERGY_GATE1) + 1);
+        } else
+        if ((option >= P159_OUTPUT_MOVING_DISTANCE_ENERGY_GATE1) && (option <= P159_OUTPUT_MOVING_DISTANCE_ENERGY_GATE8)) {
+          options[option] = concat(Plugin_159_valuename(P159_OUTPUT_MOVING_DISTANCE_GATE_index, true),
+                                   (option - P159_OUTPUT_MOVING_DISTANCE_ENERGY_GATE1) + 1);
+        } else {
+          options[option] = Plugin_159_valuename(option, true);
+        }
+      }
+
+      const int valueCount = P159_NR_OUTPUT_VALUES;
+
+      for (uint8_t i = 0; i < valueCount; ++i) {
+        const uint8_t pconfigIndex = i + P159_QUERY1_CONFIG_POS;
+        sensorTypeHelper_loadOutputSelector(event, pconfigIndex, i, optionCount, options);
+      }
+      success = true;
+
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      addFormSelector_YesNo(F("Engineering mode"), F("eng"), P159_ENGINEERING_MODE, true);
+      addFormNote(F("When changing this setting the page will be reloaded"));
+      success = true;
+
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      P159_ENGINEERING_MODE = getFormItemInt(F("eng"));
+
+      success = true;
+
+      break;
+    }
+    case PLUGIN_INIT:
+    {
+      int8_t rxPin               = serialHelper_getRxPin(event);
+      int8_t txPin               = serialHelper_getTxPin(event);
+      ESPEasySerialPort portType = serialHelper_getSerialType(event);
+
+      // Create the P159_data_struct object that will do all the sensor interaction
+      initPluginTaskData(event->TaskIndex, new (std::nothrow) P159_data_struct(portType, rxPin, txPin, P159_ENGINEERING_MODE == 1));
+      P159_data_struct *P159_data = static_cast<P159_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+      success = nullptr != P159_data;
+      addLog(LOG_LEVEL_INFO, concat(F("P159: INIT, success: "), success ? 1 : 0));
+
+      break;
+    }
+
+    case PLUGIN_READ:
+    {
+      P159_data_struct *P159_data = static_cast<P159_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+      if (nullptr != P159_data) {
+        success = P159_data->plugin_read(event);
+      }
+
+      break;
+    }
+
+    case PLUGIN_WRITE:
+    {
+      // TODO
+
+      break;
+    }
+
+    case PLUGIN_EXIT:
+    {
+      P159_data_struct *P159_data = static_cast<P159_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+      if (nullptr != P159_data) {
+        P159_data->disconnectSerial();
+      }
+
+      success = true;
+
+      break;
+    }
+
+    case PLUGIN_TEN_PER_SECOND:
+    {
+      P159_data_struct *P159_data = static_cast<P159_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+      if (nullptr != P159_data) {
+        success = P159_data->processSensor(); // This has 50 msec delays included, so should NOT go in PLUGIN_FIFTY_PER_SECOND !
```

Comment:
50 msec delay?
Sounds like we should split this into a set and read out call...
50 msec delay is not even working when you have 2 instances running.